File: app/src/main/java/com/maciega/bartosz/crawl/CrawlLogAdapter.java
package com.maciega.bartosz.crawl;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by bartoszmaciega on 17/10/16.
 */

public class CrawlLogAdapter extends BaseAdapter {

    private Context context;
    private List<String> urls = new ArrayList<>();


    public CrawlLogAdapter(Context context) {
        this.context = context;
    }


    public void setData(List<String> urls) {
        this.urls = urls;
        notifyDataSetChanged();
    }

    public void clear(){
        this.urls.clear();
        notifyDataSetChanged();
    }


    public void addData(String url) {
        this.urls.add(url);
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return urls.size();
    }

    @Override
    public Object getItem(int position) {
        return urls.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        String item = (String) getItem(position);
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.crawl_log_item,parent,false);
        TextView url = (TextView) view.findViewById(R.id.url);
        url.setText(item != null? item : "empty");
        return view;
    }
}

File: app/src/main/java/com/maciega/bartosz/crawl/adapter/CrawlLogAdapter.kt
package com.maciega.bartosz.crawl.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.maciega.bartosz.crawl.R

import java.util.ArrayList

/**
 * Created by bartoszmaciega on 17/10/16.
 */

class CrawlLogAdapter(private val context: Context) : BaseAdapter() {
    private var urls: MutableList<String> = ArrayList()


    fun setData(urls: MutableList<String>) {
        this.urls = urls
        notifyDataSetChanged()
    }

    fun clear() {
        this.urls.clear()
        notifyDataSetChanged()
    }


    fun addData(url: String) {
        this.urls.add(url)
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return urls.size
    }

    override fun getItem(position: Int): Any {
        return urls[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        val item = getItem(position) as String?
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.crawl_log_item, parent, false)
        val url = view.findViewById(R.id.url) as TextView
        url.text = item ?: "empty"
        return view
    }
}
